service/src/test/java/com/ge/predix/acs/service/policy/admin/PolicyManagementServiceTest.java

```diff
@@ -163,6 +164,106 @@ public void testCreateApmPolicySetPositive() {
+                .deserializeFromFile("obligation/set-with-1-policy-bad-action-template-null.json", PolicySet.class);
+        this.policyService.upsertPolicySet(policySet);
+    }
+
+    @Test(
+            expectedExceptions = { PolicyManagementException.class },
+            expectedExceptionsMessageRegExp = ".*Obligation validation failed\\. actionArgument.*cannot be null "
+                    + "or blank on .*")
+    public void testValidatePolicyBadObligationActionArgumentNameBlank() {
+        PolicySet policySet = this.jsonUtils.deserializeFromFile(
+                "obligation/set-with-1-policy-bad-obligation-action-argument-name-blank.json", PolicySet.class);
+        this.policyService.upsertPolicySet(policySet);
+
+    }
+
+    @Test(
+            expectedExceptions = { PolicyManagementException.class },
+            expectedExceptionsMessageRegExp = ".*Obligation validation failed\\. actionArgument.*cannot be null .*")
+
+    public void testValidatePolicyBadObligationActionArgumentNameNull() {
+        PolicySet policySet = this.jsonUtils.deserializeFromFile(
+                "obligation/set-with-1-policy-bad-obligation-action-argument-name-null.json", PolicySet.class);
+        this.policyService.upsertPolicySet(policySet);
+    }
+
+    @Test(
+            expectedExceptions = { PolicyManagementException.class },
+            expectedExceptionsMessageRegExp = ".*Obligation validation failed\\. actionArguments names cannot"
+                    + " be repeated on  index.*")
+    public void testValidatePolicyBadObligationActionArgumentDuplicatedNames() {
+        PolicySet policySet = this.jsonUtils.deserializeFromFile(
+                "obligation/set-with-1-policy-bad-obligation-action-argument-name-duplicated.json", PolicySet.class);
+        this.policyService.upsertPolicySet(policySet);
+    }
+
+    @Test(
+            expectedExceptions = { PolicyManagementException.class },
+            expectedExceptionsMessageRegExp = ".*Policy validation failed\\. The following obligationsIds  "
+                    + "does not match with the obligations defined for this policy set.*")
+    public void testValidatePolicyObligationIdsNotFound() {
+        PolicySet policySet = this.jsonUtils.deserializeFromFile(
+                "obligation/set-with-1-policy-bad-obligations-ids-not-found.json", PolicySet.class);
+        this.policyService.upsertPolicySet(policySet);
+    }
+
+    @Test(
+            expectedExceptions = { PolicyManagementException.class },
+            expectedExceptionsMessageRegExp = ".*Obligation validation failed\\. Obligation ids need to unique\\."
+                    + " Repeated values.*")
+    public void testValidateObligationIdsNotUnique() {
+        PolicySet policySet = this.jsonUtils
+                .deserializeFromFile("obligation/set-with-1-policy-bad-obligation-ids-repeated.json", PolicySet.class);
+        this.policyService.upsertPolicySet(policySet);
+    }
+
```

Comment:
Add positive and negative tests for:
- Testing the validity of the `type` of obligation
- Testing the validity of the `optional` attribute
- Testing the validity of the `actionTemplate.select` list in the presence of duplicated `name`s (might be an edge case?)
- Testing the validity of the `actionTemplate.filter` attribute in the presence of placeholders that don't get expanded since they're absent from the `actionArguments` list

Reply:
Type is enum, other values rather than  CUSTOM, WILL throw a 400
Optional has been removes.

Testing the validity of the actionTemplate.select list in the presence of duplicated names (might be an edge case?)
Don't understand this action template is any object, can contain a collection or not

Testing the validity of the actionTemplate.filter attribute in the presence of placeholders that don't get expanded since they're absent from the actionArguments list

Don't understand this action template is any object, can contain a collection or not

Reply:
Since action template is a free object, we don't know what to validate against, we can consider in the future a mechanism to upload a contract specification for the action template.

Reply:
double checking that we are testing that action arguments are not repeated.

Reply:
All these tests could be moved to PolicySetValidatorTest